splice_machine/src/main/java/com/splicemachine/derby/ddl/DDLUtils.java

```diff
@@ -376,36 +367,45 @@ public static void preAlterStats(DDLMessage.DDLChange change, DataDictionary dd,
     public static void preDropSchema(DDLMessage.DDLChange change, DataDictionary dd, DependencyManager dm) throws StandardException {
         if (LOG.isDebugEnabled())
             SpliceLogUtils.debug(LOG,"preDropSchema with change=%s",change);
-        dd.getDataDictionaryCache().schemaCacheRemove(change.getDropSchema().getSchemaName());
-        dd.getDataDictionaryCache().oidSchemaCacheRemove(ProtoUtil.getDerbyUUID(change.getDropSchema().getSchemaUUID()));
+        DDLMessage.DropSchema dropSchema = change.getDropSchema();
+        UUID schemaId = ProtoUtil.getDerbyUUID(dropSchema.getSchemaUUID());
+        try {
+            try (SpliceTransactionResourceImpl transactionResource = new SpliceTransactionResourceImpl()) {
+                TxnView txn = DDLUtils.getLazyTransaction(change.getTxnId());
+                transactionResource.marshallTransaction(txn);
+                SchemaDescriptor sd = dd.getSchemaDescriptor(schemaId, transactionResource.getLcc().getTransactionExecute());
+                if(sd==null) // Schema Descriptor transaction never committed
+                    return;
+                dm.invalidateFor(sd,DependencyManager.DROP_SCHEMA,transactionResource.getLcc());
```

Comment:
Looks like an additional logic besides refactoring right? Could you explain why this is necessary?

Reply:
This is true. We found out with Daniel that the code in `preDropSchema` did not invalidate potential dependencies to that schema. In a pre multidb world, it probably does not matter, but we should do in `preDropSchema` what we do in `preDropTable` anyway

Reply:
OK, thanks for the clarification.